src/nodes/input-battery.html

```diff
@@ -21,37 +26,45 @@
     RED.nodes.registerType('victron-battery', {
         category: 'input',
         defaults: {
-            service: {value: ""}
+            service: {value: ""},
+            path: {value: ""}
         },
         color: '#4790d0',
         inputs: 0,
         outputs: 1,
 
         icon: "bridge.png",
         label: function() {
-            let label = this.service
-            ? ' - ' + JSON.parse(this.service).label
-            : '';
+            let svc = JSON.parse(this.service).name
+            let path = JSON.parse(this.path).label
 
-            return this.name || `Battery${label}`;
+            return this.name || `Battery - ${svc} - ${path}`;
         },
         oneditprepare: function() {
             let _this = this;
             $.getJSON('victron/services/battery', function(data) {
-                if (Object.keys(data).length !== 0) {
-                    let selectElem = $('#node-input-service');
-                    selectElem.val();
+                if (data.length !== 0) {
+                    let serviceElem = $('#node-input-service');
+                    let pathElem = $('#node-input-path');
+                    serviceElem.val();
+                    pathElem.val();
+            
+                    data.forEach(monitor => {
+                        serviceElem.append(
+                            $('<option/>', {value: JSON.stringify(monitor), text: monitor.name})
+                        );
+                    });
 
-                    let theBattery = data[Object.keys(data)[0]];
+                    let availablePaths = data[0]['paths'];
```

Comment:
Add a comment / TODO here?

Reply:
The REST API (and the battery node as well) already supports for individual services per device. I will implement an onchange handler to change the measurement select to mirror the selected service.